src/index.ts

```diff
@@ -161,6 +161,31 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
       // so remove the header from the response we'll return
       headers.delete('cache-control')
     }
+
+    // Several preconditions must be met for a 304 Not Modified:
+    // - mime type cannot be HTML. there several reasons for this
+    // but the most glaring is that CF uses chunked transfer encoding
+    // when inserting html via cache api. Thus the body is streamed
+    // before the full file contents are known
+    // - client sends if-none-match
+    // - resource has etag
+    // - test if-none-match against etag
+    let shouldRevalidate = [
+      mimeType.indexOf('html') !== -1, // prevents falsy values
+      request.headers.has('if-none-match'),
+      response.headers.has('etag'),
+      request.headers.get('if-none-match') === response.headers.get('etag'),
+    ].every((val) => val === true)
```

Comment:
Just wanted to quickly say this entire PR is gold and appreciate your efforts here.

This section specifically, probably saved me hours and really opened my eyes to how caching can work / be leveraged on CF. I think this would be insanely helpful to see this concept in the template gallery after all this is done